src/indexer/utils/evm/index.ts

```diff
@@ -196,16 +197,22 @@ export async function saveDepositLogs(
   transferRepository: TransferRepository,
   depositRepository: DepositRepository,
   transferMap: Map<string, string>,
+  ofacComplianceService: OfacComplianceService,
 ): Promise<void> {
   let transfer = await transferRepository.findTransfer(decodedLog.depositNonce, Number(decodedLog.fromDomainId), Number(decodedLog.toDomainId))
+
+  const { sender } = decodedLog
+
+  const senderStatus = await ofacComplianceService.checkSanctionedAddress(sender)
+
   if (!transfer) {
-    transfer = await transferRepository.insertDepositTransfer(decodedLog)
+    transfer = await transferRepository.insertDepositTransfer(decodedLog, senderStatus)
```

Comment:
I would not send sender status in this and just save the account before going into insert deposit transfer.